File: codingGame/horse-racing/horse-racing.go
package main

import (
	"fmt"
	"sort"
)

func main() {
	var N int
	fmt.Scan(&N)
	horses := make([]int, N)

	for i := 0; i < N; i++ {
		fmt.Scan(&horses[i])
	}
	sort.Ints(horses)
	prev := -1
	minDiff := horses[1] - horses[0]
	for _, h := range horses {
		current := h - prev
		if prev != -1 && minDiff > current {
			minDiff = h - prev
		}

		prev = h
	}
	fmt.Println(minDiff)
}

File: codingGame/horse-racing/horse-racing.js
var N = parseInt(readline()),
    pi = [],
    prev = false,
    minDiff;
for (var i = 0; i < N; i++) {
    pi.push(parseInt(readline()));
}

for (var p of pi.sort(function(a, b) { return a - b; })) {
    if (!prev || minDiff > p - prev)
        minDiff = p - prev;

    prev = p;
}
print(minDiff);